### PR TITLE
fixes test cases that are failing in windows for archive file

### LINKED ISSUE #334: Unit Test failure for windows OS in archive.go file
### Please fill out the issue checklist below and provide ALL the requested information.

- [x] I reviewed open and closed Github issues that may be related to my problem.
- [x] I am reporting a bug that others will be able to reproduce.

### Describe the bug
Unit test failing in Windows OS for archive.go file test cases
OS: Windows

Error details 

`--- FAIL: TestHandleZip (0.02s)
	--- FAIL: TestHandleZip/valid_jar (0.01s)
    	archive_test.go:155: ExtractZip() (-want, +got) =   bytes.Join({
        	-   "package main\r",
        	-   "\r",
        	-   "import \"fmt\"\r",
        	-   "\r",
        	-   "func main() {\r",
        	-   "\tfmt.Println(\"hello world\")\r",
        	-   "}\r",
        	+   "package main",
        	+   "",
        	+   `import "fmt"`,
        	+   "",
        	+   "func main() {",
        	+   `   	fmt.Println("hello world")`,
        	+   "}",
            	"",
          	}, "\n")
	--- FAIL: TestHandleZip/valid_zip (0.00s)
    	archive_test.go:155: ExtractZip() (-want, +got) =   bytes.Join({
        	-   "package main\r",
        	-   "\r",
        	-   "import \"fmt\"\r",
        	-   "\r",
        	-   "func main() {\r",
        	-   "\tfmt.Println(\"hello world\")\r",
        	-   "}\r",
        	+   "package main",
        	+   "",
        	+   `import "fmt"`,
        	+   "",
        	+   "func main() {",
        	+   `   	fmt.Println("hello world")`,
        	+   "}",
            	"",
          	}, "\n")`
### Expected behavior

All unit test cases are successful

### Steps to Reproduce

clone apps-cli-plugin code git clone https://github.com/vmware-tanzu/apps-cli-plugin in windows and run go test ./....

### Version (Apps plugin version, Version of K8s running on cluster)
Put the output of the following commad
```sh
tanzu version && tanzu apps version
```

### PR DESCRIPTION
**Pull request by shaheerkootteeri**
Pull request

### What this PR does / why we need it

PR fixes test cases that are failing in windows for archive file due the line break differences in different oS.

### Which issue(s) this PR fixes

Fixes #334 

### Describe testing done for PR

Validated executing unit test cases in mac as well as windows OS, also installing them.

### CODE REVIEW

File: pkg/source/archive_test.go [hunk 1]
```diff
@@ -162,3 +165,11 @@ func TestHandleZip(t *testing.T) {
 		})
 	}
 }
+
+func normalizeNewlines(d []byte) []byte {
+	// replace CR LF \r\n (windows) with LF \n (unix)
+	d = bytes.Replace(d, []byte{13, 10}, []byte{10}, -1)
```
[the remark this answers is not in the document]
Comment (shashwathi) [reply pkg/source/archive_test.go:171]: same for `/r`

[the remark this answers is not in the document]
Comment (shaheerkootteeri) [reply pkg/source/archive_test.go]: Updated

[the remark this answers is not in the document]
Comment (warango4) [reply pkg/source/archive_test.go]: Please resolve conversations when done

**Review by odinnordico — APPROVED** [review]

File: pkg/source/archive_test.go [hunk 2]
```diff
@@ -162,3 +163,13 @@ func TestHandleZip(t *testing.T) {
 		})
 	}
 }
+
+func normalizeNewlines(d []byte) []byte {
+	CR := []byte{13}
```
Comment (shashwathi) [inline pkg/source/archive_test.go:168]: declare linebreak as a constant and also as `/n` instead of byte array. It makes the code more readable

**Review by shashwathi — APPROVED** [review]

**Pull request merged**